lib/ansible/plugins/grpc/__init__.py

```diff
@@ -0,0 +1,52 @@
+# (c) 2019 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.plugins import AnsiblePlugin
+
+
+class GrpcBase(AnsiblePlugin):
+    """
+    A base class for implementing gRPC abstraction layer
+    """
+
+    __rpc__ = ['channel', 'get_config', 'edit_config', 'get']
+
+    def __init__(self, connection):
+        super(GrpcBase, self).__init__()
+        self._connection = connection
+
+    @property
+    def channel(self):
+        return self._connection._channel
+
+    def get_config(self, section=None):
```

Comment:
```suggestion
    @abstractmethod
    def get_config(self, section=None):
```

Reply:
The grpc protocol does not define any specific api's or RPC as in the case of netconf and restconf. The gRPC api's support for any platform is defined in the protobuf files, that's the reason these methods are not defined as `abstractmethod` as some of the platform may not have the need to implement these api's

Reply:
I guess if these methods aren't guaranteed to exist, then why present them in the base class at all? What is the benefit of having this method exist but be unimplemented on an implementation that doesn't use it?